subsys/net/lib/sockets/sockets.c

```diff
@@ -765,6 +766,13 @@ static inline ssize_t zsock_recv_dgram(struct net_context *ctx,
 		return -1;
 	}
 
+#if defined(CONFIG_NET_PKT_RXTIME_STATS)
```

Comment:
... that will help to avoid these ifdef. (and the line will be much shorter as well).